sdk/go/common/workspace/templates.go

```diff
@@ -697,16 +707,41 @@ func newExistingFilesError(existing []string) error {
 	return errors.New(message)
 }
 
+type TemplateNotFoundError struct {
+	templateName string
+	suggestions  []string
+}
+
+func (err TemplateNotFoundError) Error() string {
+	message := fmt.Sprintf("template '%s' not found", err.templateName)
+	if len(err.suggestions) == 0 {
+		return message
+	}
+	// Build-up error message with suggestions.
+	message = message + "\n\nDid you mean this?\n"
+	for _, suggestion := range err.suggestions {
+		message = message + fmt.Sprintf("\t%s\n", suggestion)
+	}
+
+	return message
+}
+
+func (TemplateNotFoundError) Is(target error) bool {
+	_, v := target.(TemplateNotFoundError)
+	_, p := target.(*TemplateNotFoundError)
+	return v || p
+}
```

Comment:
Do we really need to do this for every error type we add?

Reply:
If we want `errors.Is` to work *and* if `==` doesn't do the job (`TemplateNotFoundError` is not comparable). Instead of implementing `Is`, we could instead use `errors.As` in our check.

Would you prefer that?

Reply:
I have switched to `As` and deleted the `Is` block.